fili-core/src/main/java/com/yahoo/bard/webservice/web/TableFullViewProcessor.java

```diff
@@ -97,7 +97,7 @@ public TableGrainView formatTableGrain(LogicalTable logicalTable, String grain,
         resultRow.put("name", grain);
         resultRow.put("longName", StringUtils.capitalize(grain));
         resultRow.put("description", "The " + logicalTable.getName() + " " + grain + " grain");
-        resultRow.put("retention", logicalTable.getRetention().toString());
+        resultRow.put("retention", logicalTable.getRetention() != null ? logicalTable.getRetention().toString() : "");
```

Comment:
This won't do what you think it do now that we're returning an `Optional` (which won't be `null`). We probably want `logicalTable.getRetention().map(ReadablePeriod::toString).orElse("")`

Reply:
Nevermind, this is a `LogicalTable`, not a `LogicalTableName`.